client/src/nl/sensorlab/videowall/animation/baseanimations/bouncypixels/BouncyPixel.java

```diff
@@ -0,0 +1,92 @@
+package nl.sensorlab.videowall.animation.baseanimations.bouncypixels;
+
+import java.util.ArrayList;
+
+import nl.sensorlab.videowall.animation.BaseAnimation;
+import processing.core.PGraphics;
+import processing.core.PVector;
+
+public class BouncyPixel {
+
+	BouncyPixelSystem parent;
+
+	public PVector position;
+	public PVector velocity;
+	public float diameter;
+	public int id;
+	private int color;
+
+	public BouncyPixel(BouncyPixelSystem parent, float x, float y, float d, int id, int color) {
+		this.parent = parent;
+		this.position = new PVector(x, y);
+		this.velocity = new PVector(0, 0);
+		this.diameter = d;
+		this.id = id;
+		this.color = color;
+	}
+
+	public void collide(ArrayList<BouncyPixel> bouncypixels) {
+		for (int i = 0; i < bouncypixels.size(); i++) {
+			// Ignore self
+			if(i != id){
+				BouncyPixel b = bouncypixels.get(i);
+				float dx = b.position.x - position.x;
+				float dy = b.position.y - position.y;
+
+				// The minimum distance
+				float minDistance = b.diameter/2 + diameter/2;
+
+				// The actual distance
+				float distance = (float)Math.sqrt(dx*dx + dy*dy);
+
+				// If exceeds minimum distance
+				if(distance < minDistance) {
+					float angle = (float)Math.atan2(dy, dx);
+					float targetX = position.x + (float)Math.cos(angle) * minDistance;
+					float targetY = position.y + (float)Math.sin(angle) * minDistance;
+					
+					float ax = (targetX - b.position.x) * parent.spring;
+					float ay = (targetY - b.position.y) * parent.spring;
+
+					// Update velocity
+					velocity.x -= ax;
+					velocity.y -= ay;
+
+					// Update b (change direction)
+					b.velocity.x += ax;
+					b.velocity.y += ay;
+				}		
+			}
+		}
+	}
+
+	public void update() {
```

Comment:
This method should also use delta time to decouple animation speed and frame rate